hack/schedule-parser/schedule-parser.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type PatchSchedule struct {
+	Schedules []Schedules `yaml:"schedules"`
+}
+type PreviousPatches struct {
+	Release            string `yaml:"release"`
+	CherryPickDeadline string `yaml:"cherryPickDeadline"`
+	TargetDate         string `yaml:"targetDate"`
+}
+type Schedules struct {
+	Release            string            `yaml:"release"`
+	Next               string            `yaml:"next"`
+	CherryPickDeadline string            `yaml:"cherryPickDeadline"`
+	TargetDate         string            `yaml:"targetDate"`
+	EndOfLifeDate      string            `yaml:"endOfLifeDate"`
+	PreviousPatches    []PreviousPatches `yaml:"previousPatches"`
+}
+
+var filePath string
+
+func init() {
+	flag.StringVar(&filePath, "schedule-path", "../../../sig-release/releases/schedule.yaml", "path where the schedule yaml file is located")
+}
+
+func main() {
+	flag.Parse()
+
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		log.Errorf("failed to read the file: %v", err.Error())
+		os.Exit(1)
+	}
+
+	var patchSchedule PatchSchedule
+
+	err = yaml.Unmarshal([]byte(data), &patchSchedule)
+	if err != nil {
+		log.Errorf("failed to decode the file: %v", err.Error())
+		os.Exit(1)
+	}
+
+	var output []string
+	output = append(output, "### Timeline\n")
+	for _, releaseSchedule := range patchSchedule.Schedules {
+		output = append(output, fmt.Sprintf("### %s\n", releaseSchedule.Release))
+		output = append(output, fmt.Sprintf("Next patch release is **%s**\n", releaseSchedule.Next))
+		output = append(output, fmt.Sprintf("End of Life for **%s** is **%s**\n", releaseSchedule.Release, releaseSchedule.EndOfLifeDate))
+
+		output = append(output, "| Patch Release | Cherry-picks deadline | Target date |\n| --- | --- | --- |")
+		output = append(output, fmt.Sprintf("| %s | %s | %s |", releaseSchedule.Next, releaseSchedule.CherryPickDeadline, releaseSchedule.TargetDate))
```

Comment:
I suggest using `tabwriter` to align the content: https://golang.org/pkg/text/tabwriter